builtin/providers/aws/resource_aws_db_event_subscription_test.go

```diff
@@ -0,0 +1,179 @@
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No RDS Event Subscription is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).rdsconn
+
+		opts := rds.DescribeEventSubscriptionsInput{
+			SubscriptionName: aws.String(rs.Primary.ID),
+		}
+
+		resp, err := conn.DescribeEventSubscriptions(&opts)
+
+		if err != nil {
+			return err
+		}
+
+		if len(resp.EventSubscriptionsList) != 1 ||
+			*resp.EventSubscriptionsList[0].CustSubscriptionId != rs.Primary.ID {
+			return fmt.Errorf("RDS Event Subscription not found")
+		}
+
+		*v = *resp.EventSubscriptionsList[0]
+
+		return nil
+	}
+}
+
+func testAccCheckAWSDBEventSubscriptionAttributes(v *rds.EventSubscription) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		if *v.Enabled != true {
+			return fmt.Errorf("bad enabled: %#v", *v.Enabled)
+		}
+
+		if *v.SourceType == "db-instance" {
+			return fmt.Errorf("bad source_type: %#v", *v.SourceType)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckAWSDBEventSubscriptionDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).rdsconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_db_event_subscription" {
+			continue
+		}
+
+		var err error
+		resp, err := conn.DescribeEventSubscriptions(
+			&rds.DescribeEventSubscriptionsInput{
+				SubscriptionName: aws.String(rs.Primary.ID),
+			})
+
+		if ae, ok := err.(awserr.Error); ok && ae.Code() == "SubscriptionNotFound" {
+			continue
+		}
+
+		if err == nil {
+			if len(resp.EventSubscriptionsList) != 0 &&
+				*resp.EventSubscriptionsList[0].CustSubscriptionId == rs.Primary.ID {
+				return fmt.Errorf("Event Subscription still exists")
+			}
+		}
+
+		// Verify the error
+		newerr, ok := err.(awserr.Error)
+		if !ok {
+			return err
+		}
+		if newerr.Code() != "SubscriptionNotFound" {
+			return err
+		}
+	}
+
+	return nil
+}
+
+var testAccAWSDBEventSubscriptionConfig = `
+resource "aws_sns_topic" "aws_sns_topic" {
+  name = "tf-acc-test-rds-event-subs-sns-topic"
+}
+
+resource "aws_db_event_subscription" "bar" {
+  name = "tf-acc-test-rds-event-subs2"
+  sns_topic = "${aws_sns_topic.aws_sns_topic.arn}"
+  enabled = true
+  source_type = "db-instance"
+  event_categories = [
+    "availability",
+    "backup",
+    "creation",
+    "deletion",
+    "maintenance"
+  ]
+  tags {
+    Name = "name"
+  }
+}
+`
+
+var testAccAWSDBEventSubscriptionConfigUpdate = `
+resource "aws_sns_topic" "aws_sns_topic" {
+  name = "tf-acc-test-rds-event-subs-sns-topic"
+}
+
+resource "aws_db_event_subscription" "bar" {
+  name = "tf-acc-test-rds-event-subs2"
+  sns_topic = "${aws_sns_topic.aws_sns_topic.arn}"
+  enabled = true
+  source_type = "db-instance"
+  event_categories = [
+    "availability",
+    "backup",
+    "creation",
+    "deletion",
+    "maintenance"
+  ]
+  tags {
+    Name = "new-name"
+  }
```

Comment:
We did have many cases in the past, in other resources where some fields were not updated properly and tests didn't catch this.

If we have a single acceptance test for testing update I think we should try to test as many fields as we can in that single test (not just `tags`) - i.e. can you try to make all fields which are not `ForceNew` different from `testAccAWSDBEventSubscriptionConfig` so we can verify that updates for all/most fields are working as expected?